### PR TITLE
Almost finished KafkaProducer and KafkaConsumer

### PR DESCRIPTION
**Pull request by masesdevelopers**
## Description
This PR reports mainly changes in:
* Producer: almost finished KafkaProducer, adding callback
* Consumer: almost finished KafkaConsumer, adding callback and many other methods
* Added many Exceptions
* Reviewed documentation
* Upgrade to latest JCOBridge

## Related Issue
Closed #5
Update #6 
Update #7

## How Has This Been Tested?
Tested locally

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: src/net/KafkaBridge/ClientSide/BridgedClasses/Clients/Producer/Callback.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    public class Callback : CLRListener
+    {
+        /// <inheritdoc cref="CLRListener.JniClass"/>
+        public sealed override string JniClass => "org.mases.kafkabridge.clients.producer.CallbackImpl";
+
+        readonly Action<RecordMetadata, JVMBridgeException> executionFunction = null;
+        /// <summary>
+        /// The <see cref="Action{RecordMetadata, JVMBridgeException}"/> to be executed
+        /// </summary>
+        public virtual Action<RecordMetadata, JVMBridgeException> Completion { get { return executionFunction; } }
```
Comment (mariomastrodicasa) [inline src/net/KafkaBridge/ClientSide/BridgedClasses/Clients/Producer/Callback.cs:38]: Must be named OnOnCompletion

File: src/net/KafkaBridge/ClientSide/BridgedClasses/Common/Errors/ApiException.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+*  Refer to LICENSE for more information.
+*/
+
+namespace MASES.KafkaBridge.Common.Errors
+{
+    public class ApiException : KafkaException
```
Comment (mariomastrodicasa) [inline src/net/KafkaBridge/ClientSide/BridgedClasses/Common/Errors/ApiException.cs:21]: The Exceptions seems not registered at all

File: src/net/KafkaBridge/ClientSide/BridgedClasses/Clients/Producer/Callback.cs (hunk 1 above)
Comment (masesdevelopers) [reply src/net/KafkaBridge/ClientSide/BridgedClasses/Clients/Producer/Callback.cs:38]: ok

File: src/net/KafkaBridge/ClientSide/BridgedClasses/Clients/Producer/ApiException.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+*  limitations under the License.
+*
+*  Refer to LICENSE for more information.
+*/
+
+using MASES.KafkaBridge.Common.Errors;
+
+namespace MASES.KafkaBridge.Clients.Producer
+{
+    public class BufferExhaustedException : TimeoutException
```
Comment (mariomastrodicasa) [inline src/net/KafkaBridge/ClientSide/BridgedClasses/Clients/Producer/ApiException.cs:23]: The filename is not correct

File: src/net/KafkaBridge/ClientSide/BridgedClasses/Common/Errors/ApiException.cs (hunk 2 above)
Comment (masesdevelopers) [reply src/net/KafkaBridge/ClientSide/BridgedClasses/Common/Errors/ApiException.cs:21]: We will add

File: src/net/KafkaBridge/ClientSide/BridgedClasses/Clients/Producer/ApiException.cs (hunk 3 above)
Comment (masesdevelopers) [reply src/net/KafkaBridge/ClientSide/BridgedClasses/Clients/Producer/ApiException.cs:23]: Updated

**Review by mariomastrodicasa — APPROVED** [review]

**Comment by mariomastrodicasa** [comment]
CodeQL still fails: @masesdevelopers verify it or remove it from the CI

**Pull request merged**